hack/lib/version.sh

```diff
@@ -50,6 +50,13 @@ kube::version::get_version_vars() {
 
     # Use git describe to find the version based on annotated tags.
     if [[ -n ${KUBE_GIT_VERSION-} ]] || KUBE_GIT_VERSION=$("${git[@]}" describe --tags --abbrev=14 "${KUBE_GIT_COMMIT}^{commit}" 2>/dev/null); then
+      # This translates the "git describe" to an actual semver.org
+      # compatible semantic version that looks something like this:
+      #   v1.1.0-alpha.0.6+84c76d1142ea4d
+      #
+      # TODO: We continue calling this "git version" because so many
+      # downstream consumers are expecting it there.
+      KUBE_GIT_VERSION=$(echo "${KUBE_GIT_VERSION}" | sed "s/-\([0-9]\{1,\}\)-g\([0-9a-f]\{14\}\)$/.\1\+\2/")
```

Comment:
/me can't decide in semver.org if `v1.0.0-0.11-0` > `v1.0.0-0.2-0`   is `semver` is only splitting on the `.` and doing alphabetic comparision between `11-0` and `2-0` something tells me it would give the "wrong" answer.  Then again, since nothing in the world actually seems to use semver, other than people, and people would do the right thing, maybe it doesn't matter?  Ugggh